realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -100,18 +100,40 @@ public String getClassName() {
     }
 
     /**
-     * Sets a new name for this RealmObject class. This is equivalent to renaming it.
+     * Sets a new name for this RealmObject class. This is equivalent to renaming it. When {@link RealmObjectSchema#table}
+     * has a primary key, this will transfer the primary key for the new class name.
      *
      * @param className the new name for this class.
+     * @throws IllegalArgumentException if {@param className} is {@code null} or an empty string, or its length exceeds 57 characters.
      * @see RealmSchema#rename(String, String)
      */
     public RealmObjectSchema setClassName(String className) {
         checkEmpty(className);
+        if (57 < className.length()) {
```

Comment:
Doesn't it? https://github.com/realm/realm-core/blob/a38f2d69da4c300b116753cd5a8f7dafc0811ad0/src/realm/replication.cpp#L530

But not sure if the macro is enabled in the realm mode.